### PR TITLE
Support comments and empty lines, solves #41

### PR DESCRIPTION
**Pull request by juanpaucar**
@jpvillaisaza @sestrella could you take a look please?

### CODE REVIEW

File: src/System/Hapistrano.hs [hunk 1]
```diff
@@ -330,6 +331,13 @@ restartServerCommand = do
     Nothing -> return "No command given for restart action."
     Just cmd -> runCommand (host conf) cmd
 
+cleanBuildScript :: [String] -> [String]
+cleanBuildScript allScriptLines = filter (not . isCommentOrEmpty) allScriptLines
+  where
+    isCommentOrEmpty line = isEmpty line || isComment line
```
Comment (jsl) [inline src/System/Hapistrano.hs]: Does this work if there is leading space on the function?

We may want to talk about what the code is doing here. There may be a better approach, perhaps using megaparsec's code to ignore comments, or by implementing this in a different way altogether.

Comment (juanpaucar) [inline src/System/Hapistrano.hs]: Indeed,  It trims these chars `\t, \n, \r, \f, \v` as defined by is the function [isSpace](https://hackage.haskell.org/package/base-4.9.0.0/docs/Data-Char.html#v:isSpace).
I think using a parsec might be an overkill since hapistrano is pretty lightweight in dependencies.

\cc: @jpvillaisaza any thoughts on this?

Comment (jpvillaisaza) [inline src/System/Hapistrano.hs]: @jsl @juanpaucar What should happen if there's a leading space? I don't fully understand the original question.

As far as adding megaparsec as a dependency, it wouldn't be a big change in number of dependencies. Just about two additional things to install (at least with a very recent set of packages).

**Comment by jsl** [comment]
Honestly, I can't recall why having comments or spaces breaks hapistrano in the first place. Instead of creating something that makes a bash script into something that can be executed by hapistrano, should we just scp the script to the server and eval it there using bash directly?

**Comment by jsl** [comment]
Actually, in most of our use cases, I think this script is part of the code repo, so it's already on the server. Perhaps we just change so that it can execute a script on the remote server, rather than sending the commands that we find in the file?

**Comment by juanpaucar** [comment]
What about something more like capistrano or [shipit](https://github.com/shipitjs/shipit) where we write task that should be executed remotely. Although, this probably will be a big change for hapistrano

**Comment by jpvillaisaza** [comment]
I think it breaks it because it takes each line, intercalates `&&`, and then executes that as one command.

**Comment by juanpaucar** [comment]
@jpvillaisaza Exactly that's why I only wanted to filter out comment lines and empty lines. Also trimming them in the process

**Comment by mrkkrp** [comment]
Merging this as the PR is getting rather old. I'll probably refactor it and use Megaparsec though.

**Comment by jsl** [comment]
@mrkkrp I think that this wasn't merged because we were worried about [this edge case](https://github.com/stackbuilders/hapistrano/pull/44#issuecomment-255428948). Could you check it please? If it doesn't have tests to cover things like this we may want to add them. Thanks!

**Comment by mrkkrp** [comment]
I'm going to add support for comment and tests, don't worry. I'm just merging anything so we don't have hanging work before I start refactoring, etc.
